### PR TITLE
#167727641 Build out endpoints for availing and approving pending requests

### PR DESCRIPTION
**Pull request by chrismeeky**
#### What does this PR do?
Create the following endpoints : 

- `GET  /api/v1/requests`

- `PATCH  /api/v1/request`

#### Description of Task to be completed?
Build out endpoints to :

- Fetch pending requests. Route should be JWT token protected and only Managers are allowed access to the route. Managers can view only requests created by his direct reports.

- Approve pending request. Route should be JWT token protected and only Managers are allowed access to the route. Managers can only approve requests created by his direct reports.

#### How should this be manually tested?
* Clone the repo and `cd` into the project directory
* Run `yarn install` to install the project's dependency
* Ensure all environment variables are set accordingly
* Run `yarn migrate` to create all tables
* Run `yarn seed` to seed data into the database
* Run `yarn dev-start` to start the app
* To test the endpoint on _Postman_: `GET  /api/v1/requests`,  you need to select `x-www-form-urlencoded` as the _content-type_ and fill the following field
```
{
token: [User's Token],
}
```

* To test the endpoint on _Postman_ : `PATCH  /api/v1/request`,  you need to select `x-www-form-urlencoded` as the _content-type_ and fill the following field
```
{
id: [Request ID],
token: [User's Token],
}
```
## What are the relevant pivotal tracker stories?
[#167727641](https://www.pivotaltracker.com/story/show/167727641)

### CODE REVIEW

File: src/controllers/RequestController.js [hunk 1]
```diff
@@ -66,6 +66,91 @@ class RequestController {
...
+            lineManager: req.decoded.id
+          },
+          attributes: {
+            exclude: [
+              'password',
+              'isVerified',
+              'profileImage',
+              'hasProfile',
+              'rememberDetails',
+              'createdAt',
```
Comment (chrismeeky) [inline src/controllers/RequestController.js:94]: Is there a shorter way to achieve this exclusion ?

File: src/middlewares/authorization.js [hunk 2]
```diff
@@ -47,6 +47,24 @@ class Authorization {
...
+   * @returns {object} res - Response object containing an error due
+   * to unauthorized user
+   */
+  static async confirmRole(req, res, next) {
+    if (req.decoded.role !== 'Manager') {
+      return HelperMethods.clientError(res, 'Only managers can perform this action', 401);
+    }
+    next();
+  }
 }
```
Comment (Daymorelah) [inline src/middlewares/authorization.js:68]: I like this! 👍

File: src/migrations/03-create-requests.js [hunk 3]
```diff
@@ -40,6 +40,11 @@ module.exports = {
         key: 'id'
       }
     },
+    status: {
+      type: Sequelize.STRING,
```
Comment (Daymorelah) [inline src/migrations/03-create-requests.js]: Why aren't you using an `ENUM` datatype here? @NonsoAmadi10 implemented his feature using the enum data type. You can reach out to him if you have troubles implementing it.

File: src/routes/requestRoute.js [hunk 4]
```diff
@@ -12,6 +12,14 @@ const requestRoutes = app => {
     Authorization.checkToken,
     Validate.validateUserInput,
     RequestController.bookAReturnTrip);
+
+  app.get('/api/v1/requests', Authorization.checkToken,
```
Comment (Daymorelah) [inline src/routes/requestRoute.js]: Let all parameters in the `app.get` method be on a single line.

**Review by Daymorelah — CHANGES_REQUESTED** [review]

File: src/migrations/03-create-requests.js (hunk 3 above)
Comment (chrismeeky) [reply src/migrations/03-create-requests.js]: I did but it throws this error `ERROR: type "public.enum_Requests_status" does not exist`
Most of all the solutions I googled suggest I downgrade my postgres from p7 to p6

File: src/routes/requestRoute.js (hunk 4 above)
Comment (chrismeeky) [reply src/routes/requestRoute.js]: Okay

Comment (chrismeeky) [reply src/routes/requestRoute.js]: Eslint says it's too long

**Pull request merged**